src/verify.rs

```diff
@@ -31,47 +157,15 @@ impl Error for VerificationError {}
 /// Checks beacon components to see if they are valid.
 ///
 /// For unchained mode set `previous_signature` to an empty value.
-pub fn verify(
-    pk: &G1Affine,
+pub fn verify<P: Pubkey>(
+    pk: &P,
     round: u64,
     previous_signature: &[u8],
     signature: &[u8],
 ) -> Result<bool, VerificationError> {
-    let msg_on_g2 = verify_step1(round, previous_signature);
-    verify_step2(pk, signature, &msg_on_g2)
-}
-
-/// First step of the verification.
-/// Should not be used directly in most cases. Use [`verify`] instead.
-///
-/// This API is not stable.
-#[doc(hidden)]
-pub fn verify_step1(round: u64, previous_signature: &[u8]) -> G2Affine {
     let msg = message(round, previous_signature);
-    msg_to_curve(&msg)
-}
-
-/// Second step of the verification.
-/// Should not be used directly in most cases. Use [`verify`] instead.
-///
-/// This API is not stable.
-#[doc(hidden)]
-pub fn verify_step2(
-    pk: &G1Affine,
-    signature: &[u8],
-    msg_on_g2: &G2Affine,
-) -> Result<bool, VerificationError> {
-    let g1 = G1Affine::generator();
-    let sigma = match g2_from_variable(signature) {
-        Ok(sigma) => sigma,
-        Err(err) => {
-            return Err(VerificationError::InvalidPoint {
-                field: "signature".into(),
-                msg: err.to_string(),
-            })
-        }
-    };
-    Ok(fast_pairing_equality(&g1, &sigma, pk, msg_on_g2))
+    let msg_on_g2 = P::msg_to_curve(&msg);
```

Comment:
It feels like this is not only on g2 right but can also be g1, so just `msg` would fit better.